### PR TITLE
[#116941] Implement the controller action_access_control_list

### PR DESCRIPTION
**Pull request by ventilooo**

### CODE REVIEW

File: action_access_control_list/controllers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License GPL-3.0 or later (http://www.gnu.org/licenses/gpl).
```
Comment (joaoalf) [inline action_access_control_list/controllers/__init__.py]: @ventilooo LGPL please.

File: action_access_control_list/models/ir_action_access.py [hunk 2]
```diff
@@ -4,64 +4,57 @@
...
+class IrActionAccess(models.Model):
+    """Actions Access Rule"""
 
     _name = 'ir.action.access'
+    _description = __name__
 
     group_id = fields.Many2one(
-        string='Group',
-        comodel_name='res.groups',
+        'res.groups',
```
Comment (joaoalf) [inline action_access_control_list/models/ir_action_access.py:15]: @ventilooo Aesthetic changes should be avoided unless it's made to comply with an already established projet style.

**Review by joaoalf — CHANGES_REQUESTED** [review]
- Commit msg (WIP) ??
- License mismatch

A PR comment explaining what we're trying to accomplish with these changes would be nice.

File: action_access_control_list/models/ir_action_access.py (hunk 2 above)
Comment (ghost) [reply action_access_control_list/models/ir_action_access.py:15]: @joaoalf this is my change. The module is still not in production. This was done only to make the code more standard.

File: action_access_control_list/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+from odoo.api import Environment
+from odoo.http import request
+from odoo.addons.web.controllers.main import DataSet
+
+
+_logger = logging.getLogger(__name__)
+
+EXCLUDED_MODELS = {'res.users'}
+
+BASE_METHODS = {
```
Comment (joaoalf) [inline action_access_control_list/controllers/main.py:15]: @dufresnedavid I'm not sure what's your interpretation of base method but the list seems incomplete to me. What about `load`, `search_read`, `browse`, `read_group`, `name_create`, `fields_view_get`...

File: test_action_access_control_list/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# © 2017 Savoir-faire Linux
+# License GPL-3.0 or later (http://www.gnu.org/licenses/gpl).
```
Comment (joaoalf) [inline test_action_access_control_list/__manifest__.py]: @dufresnedavid GPL -> LGPL

File: action_access_control_list/models/ir_protected_action.py [hunk 5]
```diff
@@ -1,40 +1,109 @@
...
 
 
 class IrProtectedAction(models.Model):
-
-    """
-    Secured action
-    """
+    """Secured Action"""
 
     _name = 'ir.protected.action'
```
Comment (joaoalf) [inline action_access_control_list/models/ir_protected_action.py]: @dufresnedavid IMHO `ir.action.protected` would be a more standard name.

File: action_access_control_list/README.rst [hunk 6]
```diff
@@ -2,8 +2,45 @@
...
+When the action is triggered, the access will be checked against the action ACL. If the access is granted, any additionnal access will be granted for the rest of the transaction.
+
+For example, a user could validate an invoice without having access to accounting entries.
+
+
+Standard Methods
+----------------
+Standard object methods (such as read, write, create, unlink, onchange, copy, etc) are managed differently than custom object methods (actions).
+
+If a standard method is called, the access will be checked only for the model being manipulated.
```
Comment (joaoalf) [inline action_access_control_list/README.rst:35]: @dufresnedavid Looks like you're changing standard behaviour for the base methods and if it's the case, why extrapolate the module objective?

**Review by joaoalf — CHANGES_REQUESTED** [review]

File: action_access_control_list/controllers/main.py (hunk 3 above)
Comment (ghost) [reply action_access_control_list/controllers/main.py:15]: @joaoalf right, I need to add the missing methods.

File: action_access_control_list/models/ir_protected_action.py (hunk 5 above)
Comment (ghost) [reply action_access_control_list/models/ir_protected_action.py]: @joaoalf I agree

**Pull request merged**